### DISCUSSION TITLE
1.30.4

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#17522] [General]
# Notes
- Due to [#5689](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5689) it is now possible to have the correct picture, vendor, model and description for (some) white-label devices. This especially applies to TuYa devices. In case you own such device , please submit a [Wrong device picture/vendor/model/description](https://github.com/Koenkk/zigbee2mqtt/issues/new?assignees=&labels=wrong+device&template=wrong_device.yaml&title=%5BWrong+device%5D%3A+) issue.

# Improvements
- [#5625](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5625) Expose `hysteresis` for SMaBiT AV2010/32 (@sjorge)
- #17002 Support color for EssentielB EB-E14-P45-RGBW (@Koenkk)
- [#5640](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5640) Support `remote_temparture` and `displayed_temperature` for Bosch BTH-RA (@sestriel)
- [#5637](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5637) Support `indicator_mode` for Schneider Electric 41E10PBSWMZ-VW, 41E2PBSWMZ/356PB2MBTZ and 41EPBDWCLMZ/354PBDMBTZ (@DanielNagy, @linkedupbits)
- [#5645](https://github.com/Koenkk/zigbee-herdsman-converters/issues/5645) Merge Moes MS-105B with TuYa TS0601_dimmer_2 to support more features (@djbiti1)
- [#290](https://github.com/Koenkk/zigbee-OTA/pull/290) Support OTA for IKEA E2112 (@leander091)
- [#5659](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5659) Support `keypad_lockout` and `running_state` for Schneider Electric EKO07259 (@Eilivers)
- [#5654](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5654) Support additional features of Inovelli VZM31-SN 2.14 firmware (@michaelkkehoe)
- #17331 Add `carbon_monoxide` `device_class` to Home Assistant discovery (@vryzhaiev)
- [#697](https://github.com/Koenkk/zigbee-herdsman/pull/697) EZSP adapter stability improvements (@kirovilya)
- [#696](https://github.com/Koenkk/zigbee-herdsman/pull/696) Improve request queue behaviour (@slugzero)
- [#5675](https://github.com/Koenkk/zigbee-herdsman-converters/pull/56750) Support `squawk` for Climax SRAC-23B-ZBSR (@sekyHC)
- [#5676](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5676) Detect `_TZE200_shkxsgis` as Zemismart TB26-4 (@akelmanson) 
- [#5686](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5686) Detect `929003526101` as Philips 929001953101 (@k2iuh)
- [#5684](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5684) Support `action` and `keypad_lockout` for Sinopé SW2500ZB (@ericgu08)
- #17422 Support Node.js 20 (@Koenkk)
- #17363 Add unit for cover position and tilt (@Koenkk)
- #17444 Detect `KP-ACE_00.00.03.11TC` as Climax KP-23EL-ZBS-ACE (@Koenkk)
- [#5660](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5660) Support more features for LiXee ZLinky_TIC (@Mikado8231)
- [#17363](https://github.com/Koenkk/zigbee2mqtt/issues/17363) Detect Moes AM43-0.45/40-ES-EZ (`_TZE200_zah67ekd`) as TuYa TS0601_cover_4 (@Koenkk)
- [#17465](https://github.com/Koenkk/zigbee2mqtt/issues/17465) Detect `_TZE204_lsanae15` as TuYa TS0601_din_2 (@Koenkk)
- #17475  Identify `_TZ3000_4xfqlgqo` as TuYa TS0002_switch_module (@saper-2)
- [#5694](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5694) Detect `_TZ3040_wqmtjsyk` as TuYa Motion sensor (@kirovilya)
- [#5697](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5697) Expose `energy` for Sinopé SW2500ZB and other fixes (@ericgu08)
- [#5700](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5700) Expose `ac_frequency` for Develco SPLZB-131 (@RamonSmit)
- #17506 Detect `_TZE200_xu4a5rhj` as TuYa TS0601_cover_1 (@dpbarinov)

# Fixes
- [#5641](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5641) Fix `backlight_mode` values for TuYa TS130F (@jcugat)
- [#5639](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5639) Fix Xiaomi ZNCLDJ11LM position incorrect (@fenfenS)
- #17308 Disable unsupported `power_on_behavior` for more devices (@Koenkk)
- #17387 Detect `_TZ3000_dlug3kbc` as Zemismart ZIGBEE-B09-UK (@Koenkk)
- [#5661](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5661) Fix Gledopto GL-B-003P, GL-S-004P and GL-S-006P turning off with brightness 1 (@Electry)
- #14628 Fix TuYa `_TZ3210_weaqkhab` brightness control not working (@Koenkk)
- #17358 Detect TuYa `_TZ3000_5ajpkyq6` to TS0004_switch_module (@Koenkk)
- #16709 Fix false 0 `power`/`voltage`/`current` measurements for TuYa TS011F_plug_1 (`_TZ3000_gvn91tmx`/`_TZ3000_amdymr7l`) (@Koenkk)
- [#5679](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5679) Fix `action`/`operation_mode` for Niko 552-721X2 (@sjorge)
- [#5681](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5681) Fix `occupancy`/`away_mode` issue of Namron 4512737/4512738 (@ruant)
- #17404 Enable `power_on_behavior` for Gledopto GL-C-009P (@Koenkk)
- [#17440](https://github.com/Koenkk/zigbee2mqtt/pull/17440) Typo fixes (@akx)
- [#5696](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5695) [#5696](https://github.com/Koenkk/zigbee-herdsman-converters/pull/5696) Fix Niko 552-721X1 and 552-721X2 `action` for external inputs (@svenjochems, @sjorge)

# New supported devices
This release adds support for 40 devices:
- `4058075729025` LEDVANCE SMART+ lamp E27 RGBTW (@dhanukapulkit)
- `4058075729049` LEDVANCE SMART+ lamp B22D RGBTW (@dhanukapulkit)
- `4512728` Namron Zigbee 1 channel switch K2 (Black) (@KristianAndersson)
- `5120.2210` iluminize Zigbee 3.0 actuator mini 1x 230V (@Jakobu5)
- `5123.1110` Iluminize Zigbee 3.0 controller with adjustable current 250-1500mA, max. 50W / 48V SELV (@saseba)
- `552-720X4` Niko Battery switch with 4 buttons (@jovandeginste)
- `74453` Sylvania LIGHTIFY LED soft white BR30 (@jsuisman)
- `915005987401` Philips Hue White and color ambiance Gradient Signe table lamp (black) (@kidinstinct)
- `929003056701` Philips Hue white ambiance Adore ceiling light (@Baekalfen)
- `929003553001` Philips Hue outdoor Resonate downward wall lamp (@Koenkk)
- `BMCT-SLZ` Bosch Bosch Light/shutter control unit II (@slugzero @smarts4home)
- `GL-C-103P` Gledopto Zigbee LED controller (pro) (@Koenkk)
- `HC-IWDIM-1` Heimgard Technologies Dimmer (@sekyHC)
- `HC-IWSWI-1` Heimgard Technologies In wall light switch (@sekyHC)
- `HT-MOT-2` Heimgard Technologies Motion sensor (@sekyHC)
- `HT-SLM-2` Heimgard Technologies Doorlock with fingerprint (@sekyHC)
- `K10-1220Z` TLW Global 12V LED smart driver 15W with 6-port micro plug connector (@busywait)
- `K10-1230Z` TLW Global 12V LED smart driver 30W with 6-port micro plug connector (@busywait)
- `K10-1250Z` TLW Global 12V LED smart driver 50W with 6-port micro plug connector (@busywait)
- `MAI-ZTP20F` Profalux Cover remote (@ralmn)
- `SBM300ZB4` ShinaSystem SiHAS remote control 4 button (@shinasys)
- `SBM300ZC1` ShinaSystem SiHAS remote control (@shinasys)
- `SBM300ZC2` ShinaSystem SiHAS remote control 2 button (@shinasys)
- `SBM300ZC3` ShinaSystem SiHAS remote control 3 button (@shinasys)
- `SBM300ZC4` ShinaSystem SiHAS remote control 4 button (@shinasys)
- `SM0001` TuYa Switch (@kiselio)
- `SPLZB-137` Develco Power plug (@sekyHC)
- `TS0052` TuYa Zigbee dimmer module 1 channel (@Koenkk)
- `TS011F_3_gang` TuYa 3 gang wall ac outlet (@Koenkk)
- `TS011F_with_threshold` TuYa Din rail switch with power monitoring and threshold settings (@kirovilya)
- `TS0505B_2` TuYa Zigbee RGB+CCT light (@wollo)
- `TS0601_3_phase_clamp_meter_relay` TuYa 3-phase clamp power meter with relay (@markwv)
- `TS0601_contact_temperature_humidity_sensor` TuYa Contact, temperature and humidity sensor (@CamFlyerCH)
- `TS0601_cover_4` TuYa Cover (@Koenkk)
- `TS0601_smoke_3` TuYa Photoelectric smoke detector (@pgfeller)
- `TS0601_smoke_5` TuYa Smoke sensor (@Koenkk)
- `TS0601_switch_6_gang` TuYa 6 gang switch (@willieee802)
- `TS0726` TuYa 4 gang switch with neutral wire (@evangelo18)
- `WDE002960` Schneider Electric Push button dimmer (@energizer91)
- `WXKG04LM` Xiaomi Aqara T1 double rocker wireless remote switch (@box-archived) 

<hr /><em>This discussion was created from the release <a href='https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.30.4'>1.30.4</a>.</em>

**Comment by AlexTemkin** [comment]
Z2m doesn't start after update. 
There are few records in log: 
[10:57:49] INFO: Preparing to start...
[10:57:50] INFO: Socat not enabled
[10:57:50] INFO: Starting Zigbee2MQTT...
/app/node_modules/ajv/lib/compile/validate/index.ts:144
  updateContext(it)
  ^
ReferenceError: updateContexp is not defined
    at subSchemaObjCode (/app/node_modules/ajv/lib/compile/validate/index.ts:144:3)
    at subschemaCode (/app/node_modules/ajv/lib/compile/validate/index.ts:124:7)
    at KeywordCxt.subschema (/app/node_modules/ajv/lib/compile/validate/index.ts:500:5)
    at applyPropertySchema (/app/node_modules/ajv/lib/vocabularies/applicator/properties.ts:45:11)
    at Object.code (/app/node_modules/ajv/lib/vocabularies/applicator/properties.ts:32:9)
    at keywordCode (/app/node_modules/ajv/lib/compile/validate/index.ts:532:9)
    at /app/node_modules/ajv/lib/compile/validate/index.ts:265:9
    at CodeGen.code (/app/node_modules/ajv/lib/compile/codegen/index.ts:525:33)
    at CodeGen.block (/app/node_modules/ajv/lib/compile/codegen/index.ts:680:20)
    at iterateKeywords (/app/node_modules/ajv/lib/compile/validate/index.ts:262:7)

**Reply by akx** [reply]
@AlexTemkin 

Since the error says `updateContexp` but the line of code says `updateContext()`, this sounds like a bit flip in your device's memory:

```python
>>> bin(ord('p'))
'0b1110000'
>>> bin(ord('t'))
'0b1110100'
```
